pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -419,13 +422,19 @@ impl ConnectionManagerState {
                 match res {
                     Ok(()) => Ok(()),
                     Err(e) => {
-                        use super::walreceiver_connection::ExpectedError;
-                        if e.is_expected() {
-                            info!("walreceiver connection handling ended: {e:#}");
-                            Ok(())
-                        } else {
-                            // give out an error to have task_mgr give it a really verbose logging
-                            Err(e).context("walreceiver connection handling failure")
+                        match e {
+                            WalReceiverError::SuccessfulCompletion(msg) => {
+                                info!("walreceiver connection handling ended with success: {msg}");
+                                Ok(())
+                            }
+                            WalReceiverError::ExpectedSafekeeperError(e) => {
+                                info!("walreceiver connection handling ended: {e}");
```

Comment:
```suggestion
                                info!("walreceiver connection handling ended: {e:#}");
```

Or are we sure we only want to log the top-level message?

Reply:
My earlier: https://github.com/neondatabase/neon/pull/4902#discussion_r1285712955

Reply:
Chatted, it's not an anyhow::Error, resolving.